### PR TITLE
drivers: sensor: bmg160: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

File: drivers/sensor/bmg160/bmg160_trigger.c [hunk 1]
```diff
@@ -21,15 +21,13 @@ LOG_MODULE_DECLARE(BMG160, CONFIG_SENSOR_LOG_LEVEL);
...
 
-	return gpio_pin_interrupt_configure(data->gpio,
-					    cfg->int_pin,
-					    enable
+	unsigned int flags = enable
 					    ? GPIO_INT_EDGE_TO_ACTIVE
-					    : GPIO_INT_DISABLE);
+					    : GPIO_INT_DISABLE;
+
+	return gpio_pin_interrupt_configure_dt(&cfg->int_gpio, flags);
```
Comment (gmarull) [inline drivers/sensor/bmg160/bmg160_trigger.c:30]: this change is not related, original code did not use any auxiliary var

Comment (gmarull) [reply drivers/sensor/bmg160/bmg160_trigger.c:30]: Also, this is not using the appropriate type (gpio_flags_t).

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
